### ISSUE TITLE
Empty event queue might crash the activity

### ISSUE THREAD

**Issue by JuiP** [sugarlabs/stick-hero-activity#26]
```
Traceback (most recent call last):
  File "/usr/share/sugar/activities/StickHero.activity/main.py", line 695, in make
    if event.type == pygame.KEYDOWN and event.key == 273:
UnboundLocalError: local variable 'event' referenced before assignment
```
Also reported in #24 by @quozl @srevinsaju
Don't know the exact way to reproduce this error. But happens occasionally.

**Comment by srevinsaju** [comment]
it is actually a bug within the code, the title for this PR is wrong, as the pygame event queue is not at any chance empty.

**Comment by quozl** [comment]
It is a logic error; a local variable _event_ is referenced on line 695 with the intention of it being the most recent event returned by pygame.event.get in the _for_ loop on line 346.  However, _event_ is out of scope at line 695.

Several commits in https://github.com/sugarlabs/stick-hero-activity/pull/24 did attempt to remedy this.

https://github.com/sugarlabs/stick-hero-activity/pull/27/commits/3a3e6b1631d8078b2d69e22fa921fc1a5bcf648e is the most recent attempt to remedy, and does so by creating a flag that is tested later; limiting the action to only once each time the event buffer is processed.

**Comment by srevinsaju** [comment]
@quozl you may choose to merge either of #27 or #24, as both implies the same principle. I have provided the explanation on the PR why I have implemented a different change that that provided by @ayushnawal
